easybuild/framework/easyconfig/tools.py

```diff
@@ -384,7 +384,10 @@ def stats_to_str(stats):
     txt = "{\n"
     pref = "    "
     for (k, v) in stats.items():
-        txt += "%s%s: %s,\n" % (pref, quote_str(k), quote_str(v))
+        if isyeb:
+            txt += "%s%s: %s,\n" % (pref, k, quote_special_chars(v))
+        else:
+            txt += "%s%s: %s,\n" % (pref, quote_str(k), quote_str(v))
```

Comment:
please rewrite to:

```python
for key in sorted(stats):
    if isyeb:
        val = quote_yaml_special_chars(stats[key])
    else:
        val = quote_str(stats[key])

    txt += "%s%s: %s,\n" % (pref, quote_str(key), val)
```

(this assumes `quote_str` can always be used on `key`)